packages/dd-trace/src/appsec/rasp.js

```diff
@@ -3,23 +3,102 @@
 const { storage } = require('../../../datadog-core')
 const web = require('./../plugins/util/web')
 const addresses = require('./addresses')
-const { httpClientRequestStart } = require('./channels')
+const { httpClientRequestStart, setUncaughtExceptionCaptureCallbackStart } = require('./channels')
 const { reportStackTrace } = require('./stack_trace')
 const waf = require('./waf')
+const { getBlockingAction, block } = require('./blocking')
+
+class DatadogRaspAbortError extends Error {
+  constructor (req, res, blockingAction) {
+    super('DatadogRaspAbortError')
+    this.name = 'DatadogRaspAbortError'
+    this.req = req
+    this.res = res
+    this.blockingAction = blockingAction
+  }
+}
 
 const RULE_TYPES = {
   SSRF: 'ssrf'
 }
 
-let config
+let config, abortOnUncaughtException
+
+function removeAllListeners (emitter, event) {
+  const listeners = emitter.listeners(event)
+  emitter.removeAllListeners(event)
+
+  let cleaned = false
+  return function () {
+    if (cleaned === true) {
+      return
+    }
+    cleaned = true
+
+    for (let i = 0; i < listeners.length; ++i) {
+      emitter.on(event, listeners[i])
+    }
+  }
+}
+
+function handleUncaughtExceptionMonitor (err) {
+  if (err instanceof DatadogRaspAbortError || err.cause instanceof DatadogRaspAbortError) {
```

Comment:
This `err.cause` is created by axios, it is to support axios correctly because it creates a wrapper of the Error. I think that it is not probably, but anyway I'll add a recursive check to deep into 10 levels of cause finding our error.